### PR TITLE
Enhance setup.sh email list

### LINKED ISSUE #1892: [FR] Enhanced all-in-one ./setup.sh email list (with quotas, disk usage and aliases)
# Feature Request

Display a enhanced all-in-one **_./setup.sh email list_** with quotas, disk usage and aliases for each user with one single command.

## Context

Common usage. The aim is to have more info with one glimpse. 

### Is your Feature Request related to a Problem?

No. Simply an enhancement proposal.

### Describe the Solution you'd like

The  **_./setup.sh email list_** command could display users info like this : 

```
* bobby.bob@mydomain.com ( 1,9M / ~ )
    [ aliases -> postmaster@mydomain.com, bob@mydomain.com ]

* johnny_john@mydomain.com ( 24K / 3G )
    [ aliases -> admin@mydomain.com, johnny@mydomain.com ]
```

### Are you going to implement it?

Yes, because I know the probability of someone else doing it is low and I can learn from it.

### What are you going to contribute??

Here is sample bash script which runs fine from my docker-compose working directory ; 

```
#! /bin/bash

# shellcheck source=../scripts/helper-functions.sh
#. /usr/local/bin/helper-functions.sh

#DATABASE=${DATABASE:-/tmp/docker-mailserver/postfix-virtual.cf}
DATABASE="config/postfix-accounts.cf"
ALIASES="config/postfix-virtual.cf"
QUOTAS="config/dovecot-quotas.cf"
DOCKER_MAILDATA="/var/lib/docker/volumes/docker-mailserver_maildata/_data/"
set user_found, user_data_dir, user_quota

#[[ -f ${DATABASE} ]] || errex "Error: No postfix-virtual.cf file"
#[[ -s ${DATABASE} ]] || errex "Error: Empty postfix-virtual.cf - no aliases have been added"

#grep -v "^\s*$\|^\s*\#" "${DATABASE}"
#{ grep -v "^\s*$\|^\s*\#" "${DATABASE}" || true ; } | awk -F '|' '{ print $1; }'

while read line
do
    user_found=$(echo "${line}" | awk -F '|' '{ print $1; }')
    user_data_dir=`echo "${DOCKER_MAILDATA}"$(echo "${user_found}" | awk -F '@' '{ print $2; }')/$(echo "${user_found}" | awk -F '@' '{ print $1; }')/cur`
    user_quota="~"

    # Display disk usage and quotas for specific mailbox user
    if grep -q "${user_found}" "${QUOTAS}"
    then
       user_quota=$(grep "${user_found}" "${QUOTAS}" | awk -F ':' '{ print $2; }') ;
    fi

    echo "* ${user_found} (" $(du -sh "${user_data_dir}" | awk '{ print $1; }') / "${user_quota} )"

    # Display available aliases for specific mailbox user
    echo -e "    [ aliases ->" $(grep "${user_found}" "${ALIASES}" | awk '{print $1;}' | sed ':a;N;$!ba;s/\n/, /g') "]\n"
done < "${DATABASE}"

exit 0

``` 

## Additional context

Not sure i want/have skill to "commit" directly in the project, but I think this script can be helpful (though it's certainly not perfect as it is).

### Who will that Feature be useful to?

Everyone running and managing docker-mailserver.

### What have you done already?

a sample bash script !

### PR DESCRIPTION
**Pull request by casperklein**
# Description

Display an enhanced all-in-one **_./setup.sh email list_** with quotas, disk usage and aliases for each user with one single command.

The  **_./setup.sh email list_** command output looks like this : 

```
* bobby.bob@mydomain.com ( 1,9M / ~ )
    [ aliases -> postmaster@mydomain.com, bob@mydomain.com ]

* johnny_john@mydomain.com ( 24K / 3G )
    [ aliases -> admin@mydomain.com, johnny@mydomain.com ]
```

Implements #1892

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or ENVIRONMENT.md or the documentation)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wernerfred — COMMENTED** [review]
How does the output look like if there is no quota enabled/set?

**Comment by aendeavor** [comment]
> How does the output look like if there is no quota enabled/set?

There should be no line with `[ aliases -> ...` at all as far as I can see.

**Review by aendeavor — APPROVED** [review]

**Comment by casperklein** [comment]
> How does the output look like if there is no quota enabled/set?

Like in the provided example. `~` for no quota defined: `* bobby.bob@mydomain.com ( 1,9M / ~ )`

The aliases line is only shown when there are aliases configured.

File: target/bin/listmailuser [hunk 1]
```diff
@@ -3,10 +3,39 @@
...
+  # ${QUOTA[0]} => current size
+  # ${QUOTA[1]} => configured size limit
+  # ${QUOTA[2]} => usage in percent
+  IFS=' ' read -r -a QUOTA <<< "$(doveadm quota get -u "${USER}" | tail +2 | awk '{ if ($3 == "STORAGE") { print $4" "$5" "$6 } }')"
+
+  echo "* ${USER} ( $(dovecot_quota_to_hr "${QUOTA[0]}") / $(dovecot_quota_to_hr "${QUOTA[1]}") ) [${QUOTA[2]}%]"
+
+  if [[ -f ${ALIASES} ]] && grep -q "${USER}" "${ALIASES}"
+  then
+    echo -e "    [ aliases -> $(grep "${USER}" "${ALIASES}" | awk '{print $1;}' | sed ':a;N;$!ba;s/\n/, /g')]\n"
```
Comment (wernerfred) [inline target/bin/listmailuser:35]: Worth adding a hint that the `sed ...` replacing newlines with a comma?

Just bc for me it was not clear on a first sight and as you mentioned in the Issue you needed some time too ;) 

Thinking of: 
> [x] I have commented my code, particularly in hard-to-understand areas

Other than that: LGTM 👍🏻

**Review by wernerfred — APPROVED** [review]

File: target/bin/listmailuser (hunk 1 above)
Comment (aendeavor) [reply target/bin/listmailuser:35]: I must say, the beginning of the `sed` is something I have not seen either, but I don't feel like this needs a comment, as it becomes clear that `.../\n/, /g` replaces newlines :D This is just my PoV, feel free to add it nevertheless if you feel like it's really valuable.

Comment (casperklein) [reply target/bin/listmailuser:35]: For future reference: https://stackoverflow.com/a/1252191/568737

Comment (aendeavor) [reply target/bin/listmailuser:35]: I only knew the shell-escape pattern with `$'\n'`, but this is interesting as well.

**Pull request merged**